proxy/socks/tcp.go

```diff
@@ -85,44 +71,8 @@ func (h *tcpHandler) Connect(conn core.TCPConn, target net.Addr) error {
 	if err != nil {
 		return err
 	}
-	h.Lock()
-	h.conns[conn] = c
-	h.Unlock()
-	c.SetDeadline(time.Time{})
-	go h.fetchInput(conn, c)
+	go h.handleInput(conn, c)
```

Comment:
You should propagate the Copy errors to Handle. My code shows how you can do that.